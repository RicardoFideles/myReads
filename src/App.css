html, body, .root {
  height: 100%;
}
body {
  line-height: 1.5;
}
body, .app {
  background: white;
}

/* main page */

.list-books-title {
  padding: 10px 0;
  background: #2e7c31;
  text-align: center;
}
.list-books-title h1 {
  font-weight: 400;
  margin: 0;
  color: white;
}

.list-books-content, .books-content {
  padding: 0 0 80px;
  flex: 1;
}

.bookshelf {
  padding: 0 10px 20px;
}

@media (min-width: 600px) {
  .bookshelf {
    padding: 0 20px 40px;
  }
}

.bookshelf-title {
  border-bottom: 1px solid #dedede;
}
.bookshelf-books {
  text-align: center;
}

.open-search {
  position: fixed;
  right: 25px;
  bottom: 25px;
}
.open-search a {
  display: block;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  background: #2e7d32;
  background-image: url('./icons/add.svg');
  background-repeat: no-repeat;
  background-position: center;
  background-size: 28px;
  box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
  font-size: 0;
}

/* search page */

.search-books-bar {
  position: fixed;
  width: 100%;
  top: 0;
  left: 0;
  z-index: 5;
  display: flex;
  box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 0 6px rgba(0,0,0,0.23);
}
.search-books-input-wrapper {
  flex: 1;
  background: #e9e;
}
.search-books-bar input {
  width: 100%;
  padding: 15px 10px;
  font-size: 1.25em;
  border: none;
  outline: none;
}

.close-search {
  display: block;
  top: 20px;
  left: 15px;
  width: 50px;
  height: 53px;
  background: white;
  background-image: url('./icons/arrow-back.svg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: 28px;
  font-size: 0;
}

.search-books-results {
  padding: 80px 10px 20px;
}

/* books grid */

.books-grid {
  list-style-type: none;
  padding: 0;
  margin: 0;

  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}
.books-grid li {
  padding: 10px 15px;
  text-align: left;
}

.book {
  width: 140px;
}
.book-title,
.book-authors {
  font-size: 0.8em;
}
.book-title {
  margin-top: 10px;
}
.book-authors {
  color: #999;
}

.book-top {
  position: relative;
  height: 200px;
  display: flex;
  align-items: flex-end;
}

.book-shelf-changer {
  position: absolute;
  right: 0;
  bottom: -10px;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  background: #60ac5d;
  background-image: url('./icons/arrow-drop-down.svg');
  background-repeat: no-repeat;
  background-position: center;
  background-size: 20px;
  box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
}
.book-shelf-changer select {
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
}

/* book cover */

.book-cover {
  box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
  background: #eee;
}
.book-cover-title {
  padding: 20px 10px 0;
  text-align: center;
  font-size: 0.8em;
}

.back-home {
  float:left;
  top: 20px;
  left: 15px;
  width: 50px;
  height: 53px;
  color: #FFF;
  background-image: url('./icons/arrow-back.svg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: 28px;
  font-size: 0;
}

@media (min-width: 1200px) {
  .container {
    width: 1170px;
  }
}

@media (min-width: 992px) {
  .container {
    width: 970px;
  }
}

@media (min-width: 768px) {
  .container {
    width: 750px;
  }
}

.container {
    padding-right: 15px;
    padding-left: 15px;
    margin-right: auto;
    margin-left: auto;
    margin-top : 20px;
}

.col-xs-3 {
  width: 25%;
}

.col-xs-9 {
  width: 75%;
}

.col-xs-1, .col-xs-2, .col-xs-3, .col-xs-4, .col-xs-5, .col-xs-6, .col-xs-7, .col-xs-8, .col-xs-9, .col-xs-10, .col-xs-11, .col-xs-12 {
  float: left;
}
.col-xs-1, .col-sm-1, .col-md-1, .col-lg-1, .col-xs-2, .col-sm-2, .col-md-2, .col-lg-2, .col-xs-3, .col-sm-3, .col-md-3, .col-lg-3, .col-xs-4, .col-sm-4, .col-md-4, .col-lg-4, .col-xs-5, .col-sm-5, .col-md-5, .col-lg-5, .col-xs-6, .col-sm-6, .col-md-6, .col-lg-6, .col-xs-7, .col-sm-7, .col-md-7, .col-lg-7, .col-xs-8, .col-sm-8, .col-md-8, .col-lg-8, .col-xs-9, .col-sm-9, .col-md-9, .col-lg-9, .col-xs-10, .col-sm-10, .col-md-10, .col-lg-10, .col-xs-11, .col-sm-11, .col-md-11, .col-lg-11, .col-xs-12, .col-sm-12, .col-md-12, .col-lg-12 {
  position: relative;
  min-height: 1px;
  padding-right: 15px;
  padding-left: 15px;
}