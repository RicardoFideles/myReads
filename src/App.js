import React from 'react'
import * as BooksAPI from './BooksAPI'
import Bookshelf from './Bookshelf'
import './App.css'


class BooksApp extends React.Component {
  state = {
    books : [],
    showSearchPage: false
  }

  componentDidMount () {
    BooksAPI.getAll().then((books) => {
      this.setState({books})
    })
  }


  upateBookShelf = (shelf, book) => {
    if (shelf !== 'none') {
      book.shelf = shelf
      console.log('aqui');
      this.setState((state) => ({
        books : state.books.filter((b) => b.id !== book.id).concat([book])
      }))
    }
  } 

  render() {

    const readingBooks = this.state.books.filter((book) => book.shelf === 'currentlyReading')
    const wantToReadBooks = this.state.books.filter((book) => book.shelf === 'wantToRead')
    const readBooks = this.state.books.filter((book) => book.shelf === 'read')

    return (
      <div className="app">
        {this.state.showSearchPage ? (
          <div className="search-books">
            <div className="search-books-bar">
              <a className="close-search" onClick={() => this.setState({ showSearchPage: false })}>Close</a>
              <div className="search-books-input-wrapper">
                {/*
                  NOTES: The search from BooksAPI is limited to a particular set of search terms.
                  You can find these search terms here:
                  https://github.com/udacity/reactnd-project-myreads-starter/blob/master/SEARCH_TERMS.md

                  However, remember that the BooksAPI.search method DOES search by title or author. So, don't worry if
                  you don't find a specific author or title. Every search is limited by search terms.
                */}
                <input type="text" placeholder="Search by title or author"/>

              </div>
            </div>
            <div className="search-books-results">
              <ol className="books-grid"></ol>
            </div>
          </div>
        ) : (
          <div className="list-books">
            <div className="list-books-title">
              <h1>MyReads</h1>
            </div>
            <div className="list-books-content">
              <div>
                  <Bookshelf books={readingBooks} title="Currently Reading" onUpdateBook={this.upateBookShelf}/>
                  <Bookshelf books={wantToReadBooks} title="Want to Read" onUpdateBook={this.upateBookShelf}/>
                  <Bookshelf books={readBooks} title="Read" onUpdateBook={this.upateBookShelf}/>
              </div>
            </div>
            <div className="open-search">
              <a onClick={() => this.setState({ showSearchPage: true })}>Add a book</a>
            </div>
          </div>
        )}
      </div>
    )
  }
}

export default BooksApp
