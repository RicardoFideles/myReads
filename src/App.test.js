import React from "react"
import App from "./App"
import {shallow} from "enzyme"
import Adapter from 'enzyme-adapter-react-15'
import * as Enzyme from "enzyme"

Enzyme.configure({ adapter: new Adapter() })

const fetchMock = require('fetch-mock')
fetchMock.get('*', { "books" : [{"id": 1,"title":"The Linux Command Line","subtitle":"A Complete Introduction","authors":["William E. Shotts, Jr."],"shelf":"wantToRead"},{"id": 2,"title":"Learning Web Development with React and Bootstrap","authors":["Harmeet Singh","Mehul Bhatt"],"shelf":"currentlyReading"}]})

it('renders without crashing', () => {
    expect(shallow(<App />)).toMatchSnapshot()
})

