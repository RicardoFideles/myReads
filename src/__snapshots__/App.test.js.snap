// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="app"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/search"
    render={[Function]}
  />
  <Route
    path="/book/:id"
    render={[Function]}
  />
</div>
`;
